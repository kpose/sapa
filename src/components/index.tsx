import LargeButton from './LargeButton/LargeButton';
import Started from './Started/Started';
import StartedCurrency from './StartedCurrency/StartedCurrency';
import CurrencyPicker from './CurrencyPicker/CurrencyPicker';
import FirstName from './FirstName/FirstName';
import Login from './Login/Login';
import Email from './Email/Email';
import HomeHeader from './HomeHeader/HomeHeader';
import HomeWallet from './HomeWallet/HomeWallet';
import WalletCard from './WalletCard/WalletCard';
import WavyHeader from './WavyHeader/WavyHeader';
import TabHeader from './TabHeader/TabHeader';
import Calender from './Calender/Calender';
import WalletInfo from './WalletInfo/WalletInfo';
import AddToWalletHeader from './AddToWalletHeader/AddToWalletHeader';
import AddToWalletBody from './AddToWalletBody/AddToWalletBody';
import SettingsItem from './SettingsItem/SettingsItem';
import SettingsBottom from './SettingsBottom/SettingsBottom';
import Spinner from './Spinner/Spinner';
import NetworkError from './NetworkError/NetworkError';
import ExpenseCard from './ExpenseCard/ExpenseCard';
import LoadingAnime from './Lottie/LoadingAnime';
import EmptyAnime from './Lottie/EmptyAnime';
import TransactionCard from './TransactionCard/TransactionCard';

export {
  LargeButton,
  Started,
  StartedCurrency,
  CurrencyPicker,
  FirstName,
  Login,
  Email,
  HomeHeader,
  HomeWallet,
  WalletCard,
  WavyHeader,
  TabHeader,
  Calender,
  WalletInfo,
  AddToWalletHeader,
  AddToWalletBody,
  SettingsItem,
  SettingsBottom,
  Spinner,
  NetworkError,
  ExpenseCard,
  LoadingAnime,
  EmptyAnime,
  TransactionCard,
};
