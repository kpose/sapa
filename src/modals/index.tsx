import CurrencyPicker from './CurrencyPicker/CurrencyPicker';
import AddWalletType from './AddWalletType/AddWalletType';
import AddWallet from './AddWallet/AddWallet';

export {CurrencyPicker, AddWalletType, AddWallet};
