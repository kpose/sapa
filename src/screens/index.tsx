import Welcome from './Welcome/Welcome';
import Home from './Home/Home';

export {Welcome, Home};
