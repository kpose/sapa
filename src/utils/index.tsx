import * as sizes from './fonts';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from './responsive';
import * as colors from './colors';
import {authMiddleWare} from './authMiddleware';

export {sizes, wp, hp, colors, authMiddleWare};
