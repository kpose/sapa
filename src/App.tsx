import React, {useContext, useState} from 'react';
import {store} from './redux/store';
import Routes from './navigation/routes';
import {Provider as ReduxProvider} from 'react-redux';

import {NavigationContainer} from '@react-navigation/native';
import {Provider as PaperProvider} from 'react-native-paper';
import {CombinedDarkTheme, CombinedLightTheme} from './utils/Theme';
import {ThemeContext} from './context/ThemeCotext';

const App = () => {
  const [dark, setDark] = useState(true);
  const theme = dark ? CombinedDarkTheme : CombinedLightTheme;
  const toggleTheme = () => {
    setDark(!dark);
  };

  return (
    <ThemeContext.Provider value={{theme, toggleTheme}}>
      <PaperProvider theme={theme}>
        <ReduxProvider store={store}>
          <NavigationContainer theme={theme}>
            <Routes />
          </NavigationContainer>
        </ReduxProvider>
      </PaperProvider>
    </ThemeContext.Provider>
  );
};

export default App;
