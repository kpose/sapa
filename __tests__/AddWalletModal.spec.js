import * as React from 'react';
import renderer from 'react-test-renderer';
import {fireEvent, render, waitFor} from '@testing-library/react-native';
import {AddWallet} from '../src/modals';

describe('Testing add wallet modal', () => {
  const {getByTestId, getByPlaceholderText} = render(<AddWallet />);
  const wallet_name = 'dummywallet';
  const walletInput = getByPlaceholderText('What do we call this wallet?');
  const button = getByTestId('buttontestID');

  it('expect to show wallet title ', async () => {
    await waitFor(() => {
      fireEvent.changeText(walletInput, wallet_name);
      expect(walletInput.props.value).toBe(wallet_name);

      //fireEvent.press(button);
      /* expect expect here */
    });
  });

  it('renders correctly', () => {
    const tree = renderer.create(<AddWallet />);
    expect(tree).toMatchSnapshot();
  });
});
