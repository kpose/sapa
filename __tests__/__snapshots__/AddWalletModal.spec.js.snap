// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing add wallet modal renders correctly 1`] = `
<View>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
        },
        Object {
          "textAlign": "left",
        },
        Array [
          Object {
            "fontSize": 22,
            "fontWeight": "500",
          },
          Object {
            "color": "#2A9D8F",
            "fontWeight": "bold",
            "margin": 26.5,
          },
        ],
      ]
    }
  >
    Add Wallet
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgb(231, 231, 231)",
          "borderTopLeftRadius": 4,
          "borderTopRightRadius": 4,
        },
        Object {
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "borderWidth": 0,
          "marginBottom": 40,
          "width": 600,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.26)",
          "bottom": 0,
          "height": 2,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "transform": Array [
            Object {
              "scaleY": 0.5,
            },
          ],
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "paddingBottom": 0,
            "paddingTop": 0,
          },
          Object {
            "minHeight": 133.5,
          },
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        autoFocus={true}
        editable={true}
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="What do we call this wallet?"
        placeholderTextColor="rgba(0, 0, 0, 0.54)"
        rejectResponderTermination={true}
        selectionColor="#6200ee"
        style={
          Array [
            Object {
              "flexGrow": 1,
              "margin": 0,
              "zIndex": 1,
            },
            Object {
              "paddingLeft": 12,
              "paddingRight": 12,
            },
            Object {
              "height": 133.5,
            },
            Object {
              "paddingBottom": 0,
              "paddingTop": 0,
            },
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 23,
              "fontWeight": undefined,
              "textAlign": "left",
              "textAlignVertical": "center",
            },
            false,
            Array [
              Object {},
            ],
          ]
        }
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.12)",
        "borderColor": "transparent",
        "borderRadius": 40,
        "borderStyle": "solid",
        "borderWidth": 0,
        "elevation": 0,
        "height": 80,
        "justifyContent": "center",
        "minWidth": 64,
        "width": 600,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 40,
          },
        ]
      }
      testID="buttontestID"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                false,
                Object {
                  "color": "rgba(0, 0, 0, 0.32)",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Array [
                  Object {
                    "fontSize": 17,
                    "fontWeight": "bold",
                  },
                ],
              ],
            ]
          }
        >
          Save
        </Text>
      </View>
    </View>
  </View>
</View>
`;
